src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -624,30 +624,32 @@ def _create_venv_script(
             #!{bash.path}
             set -euo pipefail
 
-            # N.B.: We convert all sandbox root relative paths to absolute paths so this script
-            # works when run with a cwd set elsewhere.
-
             # N.B.: This relies on BASH_SOURCE which has been available since bash-3.0, released in
-            # 2004. In turn, our use of BASH_SOURCE relies on the fact that this script is executed
-            # by the engine via its absolute path.
-            ABS_SANDBOX_ROOT="${{BASH_SOURCE%/*}}"
+            # 2004. It will either contain the (absolute or relative) path of the venv script, which
+            # we know to live in the sandbox root.
+            SANDBOX_ROOT="${{BASH_SOURCE%/*}}"
 
-            function ensure_absolute() {{
+            function ensure_relative_to_sandbox_root() {{
                 local value0="$1"
                 shift
                 if [ "${{value0:0:1}}" == "/" ]; then
                     echo "${{value0}}" "$@"
                 else
-                    echo "${{ABS_SANDBOX_ROOT}}/${{value0}}" "$@"
+                    # N.B.: We convert all relative paths (`./right/here`) to paths relative to the
+                    # sandbox root so this script works when run with a cwd set elsewhere. If our
+                    # sandbox root is `/this/dir` we get `/this/dir/./right/here` and if it's
+                    # `../..` (say we're in a CWD of `foo/bar` relative to the sandbox root) we get
+                    # `../.././right/here`.
```

Comment:
Hopefully more light, less heat. This now makes total sense to me, but I want to make sure it makes total sense to others to avoid further confusion.

Reply:
It does, thanks, mod the clarifications above (I wouldn't normally get this nitpicky about comment structure, but since that is the whole purpose of this change...)